File: rzops-web/src/lib/types/recycle.ts
export interface RecycleItem {
  resource_type: string;
  id: string;
  name: string;
  deleted_at: string;
  /** 删除前的完整数据快照（该表整行转 JSON） */
  data?: Record<string, unknown> | null;
}

export interface RecycleListResponse {
  data: RecycleItem[];
  count: number;
}

export interface ListRecycleQuery {
  resource_type?: string;
  q?: string;
  page?: number;
  per_page?: number;
}

export interface RestoreRequest {
  resource_type: string;
  id: string;
}

/** 回收站资源类型标签映射 */
export const resourceTypeLabels: Record<string, string> = {
  server: '服务器',
  datacenter: '数据中心',
  provider: '供应商',
  domain: '域名',
  certificate: '证书',
  server_ip: '服务器IP',
  server_port: '服务器端口',
  server_port_template: '端口模板',
  ops_site: '站点',
  database_instance: '数据库实例',
  backup_plan: '备份计划',
  monitor_target: '监控目标',
  attachment: '附件',
  contract: '合同',
  dict: '字典',
  user: '用户',
  role: '角色',
};

File: rzops-api/domain/src/models/recycle.rs
use chrono::{DateTime, Utc};
use serde::Serialize;
use uuid::Uuid;

/// 回收站条目（软删除数据的快照视图）。
#[derive(Debug, Clone, Serialize)]
pub struct RecycleEntry {
    /// 资源类型：server / datacenter / provider / domain / certificate / server_ip /
    /// server_port / server_port_template / ops_site / database_instance / backup_plan /
    /// monitor_target / attachment / contract / dict / user / role
    pub resource_type: String,
    pub id: Uuid,
    /// 展示名称（不同资源取对应名称字段）
    pub name: String,
    pub deleted_at: DateTime<Utc>,
    /// 删除前的完整数据快照（该表整行转 JSON）
    pub data: Option<serde_json::Value>,
}
